src/librustc_typeck/check/method/probe.rs

```diff
@@ -990,6 +1032,39 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
         })
     }
 
+    fn emit_unstable_name_collision_hint(
+        &self,
+        stable_pick: &Pick,
+        unstable_candidates: &[(&Candidate<'tcx>, Symbol)],
+    ) {
+        let mut diag = self.tcx.struct_span_lint_node(
+            lint::builtin::UNSTABLE_NAME_COLLISION,
+            self.fcx.body_id,
+            self.span,
+            "a method with this name will be added to the standard library in the future",
+        );
+
+        // FIXME: This should be a `span_suggestion` instead of `help`. However `self.span` only
+        // highlights the method name, so we can't use it. Also consider reusing the code from
+        // `report_method_error()`.
```

Comment:
cc @estebank regarding the span

Reply:
@kennytm You will have to modify `ProveContext` in some way to be able to store the `expr`'s span. I wouldn't mind merging this PR with the FIXME as long as a follow up ticket is filed for improving this.